### PR TITLE
v1alpha2 phase 3

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Cleaned up some leftover gce references.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: config/default/manager_credentials_patch.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ spec:
       containers:
       - name: manager
         env:
-        - name: GOOGLE_APPLICATION_CREDENTIALS
+        - name: AZURE_CREDENTIALS_FILE
```
Comment (vincepri) [inline config/default/manager_credentials_patch.yaml]: Is this correct for Azure? Do we need to pass the credentials in?

Comment (vincepri) [reply config/default/manager_credentials_patch.yaml]: Actually seems this has been removed in the other file

File: examples/generate.sh [hunk 2]
```diff
@@ -80,14 +80,25 @@ fi
...
+
+AZURE_LOCATION=${AZURE_LOCATION:?}
+export AZURE_LOCATION
+
+# Azure Credentials.
+SSH_KEY_FILE=${OUTPUT_DIR}/sshkey
+export AZURE_SUBSCRIPTION_ID_B64="$(echo -n "$AZURE_SUBSCRIPTION_ID" | base64)"
+export AZURE_TENANT_ID_B64="$(echo -n "$AZURE_TENANT_ID" | base64)"
+export AZURE_CLIENT_ID_B64="$(echo -n "$AZURE_CLIENT_ID" | base64)"
+export AZURE_CLIENT_SECRET_B64="$(echo -n "$AZURE_CLIENT_SECRET" | base64)"
```
Comment (vincepri) [inline examples/generate.sh]: You might want to `| tr -d '\n'` all base64 outputs given that on linux (which uses the gnu version) the command outputs with new lines

File: config/default/manager_credentials_patch.yaml (hunk 1 above)
Comment (CecileRobertMichon) [reply config/default/manager_credentials_patch.yaml]: which other file?

Comment (vincepri) [reply config/default/manager_credentials_patch.yaml]: `credentials.yaml`, the change shows that `credentials.json` is removed

Comment (justaugustus) [reply config/default/manager_credentials_patch.yaml]: Yeah, we probably don't need this/I haven't used this before.
We're expecting env-based configuration.

File: examples/generate.sh [hunk 3]
```diff
@@ -80,14 +80,25 @@ fi
...
+
+AZURE_RESOURCE_GROUP=${AZURE_RESOURCE_GROUP:?}
+export AZURE_RESOURCE_GROUP
+
+AZURE_LOCATION=${AZURE_LOCATION:?}
+export AZURE_LOCATION
+
+# Azure Credentials.
+SSH_KEY_FILE=${OUTPUT_DIR}/sshkey
+export AZURE_SUBSCRIPTION_ID_B64="$(echo -n "$AZURE_SUBSCRIPTION_ID" | base64 | tr -d '\n')"
```
Comment (justaugustus) [inline examples/generate.sh]: @CecileRobertMichon -- not sure that we need `| tr -d '\n'`, as `echo -n` strips the newline.

Comment (CecileRobertMichon) [reply examples/generate.sh]: @vincepri suggested to add it https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/269#discussion_r328361741

File: config/default/manager_credentials_patch.yaml [hunk 4]
```diff
@@ -9,12 +9,23 @@ spec:
...
-        - name: GOOGLE_APPLICATION_CREDENTIALS
-          value: /home/.azure/credentials.json
-        volumeMounts:
-        - name: credentials
-          mountPath: /home/.azure
-      volumes:
-      - name: credentials
-        secret:
-          secretName: manager-bootstrap-credentials
+          - name: AZURE_SUBSCRIPTION_ID
```
Comment (CecileRobertMichon) [inline config/default/manager_credentials_patch.yaml]: @justaugustus @vincepri does this make more sense?

Comment (vincepri) [reply config/default/manager_credentials_patch.yaml]: I think so, +1

File: examples/generate.sh (hunk 3 above)
Comment (justaugustus) [reply examples/generate.sh]: Sounds good!

**Pull request merged**